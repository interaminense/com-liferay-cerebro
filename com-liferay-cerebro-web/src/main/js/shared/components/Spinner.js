import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

const DISPLAYS = ['dotted', 'linear'];

const SIZES = ['sm', 'default', 'md', 'lg'];

class Spinner extends Component {
	render() {
		const {display, size} = this.props;

		const classes = getCN('spinner-root', 'loading-icon', display, {
			[`loading-icon-${size}`]: size
		});

		return <div {...this.otherProps()} class={classes} />;
	}
}

Spinner.PROPS = {
	display: Config.oneOf(DISPLAYS).value(DISPLAYS[0]),
	size: Config.oneOf(SIZES)
};

Spinner.DISPLAYS = DISPLAYS;
Spinner.SIZES = SIZES;

export default Spinner;