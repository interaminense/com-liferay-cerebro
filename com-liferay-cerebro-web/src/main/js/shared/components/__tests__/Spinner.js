import Spinner from '../Spinner';

describe('Spinner', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Spinner();

		expect(component).toMatchSnapshot();
	});

	it('should render', () => {
		component = new Spinner({
			display: 'linear',
			size: 'lg'
		});

		expect(component).toMatchSnapshot();
	});
});