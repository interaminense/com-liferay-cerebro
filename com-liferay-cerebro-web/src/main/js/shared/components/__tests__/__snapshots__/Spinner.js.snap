// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `<div class="spinner-root loading-icon dotted"></div>`;

exports[`Spinner should render 2`] = `<div class="spinner-root loading-icon linear loading-icon-lg"></div>`;
